opencog/rule-engine/forwardchainer/ForwardChainer.cc

```diff
@@ -345,58 +337,84 @@ HandleSeq ForwardChainer::apply_rule(Handle rhandle,bool search_in_focus_set /*=
 {
     HandleSeq result;
 
-    if (search_in_focus_set) {
-        //This restricts PM to look only in the focus set
-        AtomSpace focus_set_as;
+    //Check for fully grounded outputs returned by derive_rules.
+    if (not contains_atomtype(rhandle, VARIABLE_NODE)) {
+        //Subatomic matching may have created a non existing implicant
+        //atom and if the implicant doesn't exist, nor should the implicand.
+        Handle implicant = BindLinkCast(rhandle)->get_body();
+        HandleSeq hs;
+        if (implicant->getType() == AND_LINK or implicant->getType() == OR_LINK)
+            hs = LinkCast(implicant)->getOutgoingSet();
+        else
+            hs.push_back(implicant);
+        //Actual checking here.
+        for (Handle& h : hs) {
+            if (_as.get_atom(h) == Handle::UNDEFINED)
```

Comment:
Yes, even if the grounded implicant is in the atomspace, we should check that it is a member of the focus set for the case where focus set is supplied. I will add that.Thanks.